pkg/asset/agent/image/ignition.go

```diff
@@ -122,14 +130,42 @@ func (a *Ignition) Generate(dependencies asset.Parents) error {
 		},
 	}
 
-	nodeZeroIP, err := RetrieveRendezvousIP(agentConfigAsset.Config, agentHostsAsset.Hosts, agentManifests.NMStateConfigs)
-	if err != nil {
-		return err
-	}
+	clusterName := ""
+	imageTypeISO := "full-iso"
+	numMasters := 0
+	numWorkers := 0
 
-	logrus.Infof("The rendezvous host IP (node0 IP) is %s", nodeZeroIP)
+	switch agentWorkflow.Workflow {
+	case workflow.AgentWorkflowTypeInstall:
+		// Set rendezvous IP.
+		nodeZeroIP, err := RetrieveRendezvousIP(agentConfigAsset.Config, agentHostsAsset.Hosts, agentManifests.NMStateConfigs)
+		if err != nil {
+			return err
+		}
+		a.RendezvousIP = nodeZeroIP
+		logrus.Infof("The rendezvous host IP (node0 IP) is %s", a.RendezvousIP)
+		// Define cluster name and image type.
+		clusterName = fmt.Sprintf("%s.%s", agentManifests.ClusterDeployment.Spec.ClusterName, agentManifests.ClusterDeployment.Spec.BaseDomain)
+		if agentManifests.AgentClusterInstall.Spec.PlatformType == hiveext.ExternalPlatformType {
+			imageTypeISO = "minimal-iso"
+		}
+		// Fetch the required number of master and worker nodes.
+		numMasters = agentManifests.AgentClusterInstall.Spec.ProvisionRequirements.ControlPlaneAgents
+		numWorkers = agentManifests.AgentClusterInstall.Spec.ProvisionRequirements.WorkerAgents
+
+	case workflow.AgentWorkflowTypeAddNodes:
+		// In the add-nodes workflow, every node will act independently from the others.
+		a.RendezvousIP = "127.0.0.1"
+		// Reuse the existing cluster name.
+		clusterName = clusterInfo.ClusterName
+		// Fetch the required number of master and worker nodes.
+		numMasters = 0
```

Comment:
So for the release is replacing a master node not allowed? Maybe add a comment here regarding that.